drivers/pinctrl/pinctrl_wch_afio.c

```diff
@@ -39,7 +40,14 @@ int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt, uintp
 			cfg |= BIT(3);
 		} else {
 			if (pins->bias_pull_up || pins->bias_pull_down) {
+				/* "With pull up and pull down" mode */
 				cfg |= BIT(3);
+			} else if (is_adc) {
+				/* Analog input mode */
+				cfg = 0;
+			} else {
+				/* Floating input mode */
+				cfg |= BIT(0);
```

Comment:
it's the below if you want floating input?
```suggestion
				cfg |= BIT(2);
```